### PR TITLE
Add backoff_connection_errors

### PR DESCRIPTION
**Pull request by evgeny-stakewise**

### CODE REVIEW

File: sw_utils/ipfs.py [hunk 1]
```diff
@@ -238,10 +240,20 @@ async def remove(self, ipfs_hash: str) -> None:
 
 
 class IpfsFetchClient:
-    def __init__(self, endpoints: list[str], timeout: int = 60):
+    def __init__(
+            self,
+            endpoints: list[str],
+            timeout: int = 60,
```
Comment (cyc60) [inline sw_utils/ipfs.py:246]:
```suggestion
            retry_timeout: int = 60,
```

File: sw_utils/decorators.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        max_tries=max_tries,
+        max_time=max_time,
+        **kwargs
+    )
+
+
+def backoff_in_method(
+        exception: Type[Exception] | tuple[Type[Exception], ...],
+        max_tries_attr: str = None,
+        max_time_attr: str = None
```
Comment (cyc60) [inline sw_utils/decorators.py:36]:
```suggestion
        max_time_attr: str = 'retry_timeout'
```

Comment (cyc60) [reply sw_utils/decorators.py:36]: or `retry_max_time`

File: sw_utils/ipfs.py [hunk 3]
```diff
@@ -238,10 +240,20 @@ async def remove(self, ipfs_hash: str) -> None:
...
+            request_timeout: int = 60
+    ):
         self.endpoints = endpoints
         self.timeout = timeout
+        self.request_timeout = request_timeout
 
+    @backoff_in_method(
+        RuntimeError,
```
Comment (cyc60) [inline sw_utils/ipfs.py:254]: may be create custom exception? `like IPFSException`

**Review by cyc60 — CHANGES_REQUESTED** [review]

File: sw_utils/decorators.py (hunk 2 above)
Comment (evgeny-stakewise) [reply sw_utils/decorators.py]: `backoff_in_method` was deleted then

File: sw_utils/ipfs.py (hunk 3 above)
Comment (evgeny-stakewise) [reply sw_utils/ipfs.py]: Lets postpone ipfs errors handling until next PR.

File: sw_utils/ipfs.py (hunk 1 above)
Comment (evgeny-stakewise) [reply sw_utils/ipfs.py:244]: `backoff_in_method` was deleted

Comment (evgeny-stakewise) [reply sw_utils/ipfs.py:244]: I have deleted `request_timeout` param. Let's deal with ipfs in another PR.

File: sw_utils/decorators.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+
+def backoff_aiohttp_errors(
+        max_tries: int | None = None,
+        max_time: int | None = None,
+        **kwargs
+):
+    """
+    Motivation: retry only connection errors.
+    Do not retry on getting HTTP 400 or 500 errors.
```
[the remark this answers is not in the document]
Comment (evgeny-stakewise) [reply sw_utils/decorators.py]: Added 50x errors handling.

**Review by cyc60 — APPROVED** [review]

**Pull request merged**